geoapps/base/dash_application.py

```diff
@@ -300,3 +313,253 @@ def workspace(self, workspace):
+            ]
+
+        return object_options, object_value, ui_json_data, None, None
+
+    @staticmethod
+    def get_port() -> int:
+        """
+        Loop through a list of ports to find an available port.
+
+        :return port: Available port.
+        """
+        port = None
+        for p in np.arange(8050, 8101):
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                in_use = s.connect_ex(("localhost", p)) == 0
+            if in_use is False:
+                port = p
+                break
+        if port is None:
+            print("No open port found.")
+        return port
+
+    @staticmethod
+    def start_server(
+        port: int,
+        app_class: BaseDashApplication,
+        ui_json: InputFile = None,
+        ui_json_data: dict = None,
+        params: BaseParams = None,
+    ):
+        """
+        Launch dash app server using given port.
+
+        :param port: Port for where to launch server.
+        :param app_class: Type of app to create.
+        :param ui_json: ifile corresponding to the ui_json_data.
+        :param ui_json_data: Dict of current params to provide to app init.
+        :param params: Current params to pass to new app.
+        """
+        app = app_class(ui_json=ui_json, ui_json_data=ui_json_data, params=params)
+        if port is not None:
+            app.app.run_server(host="127.0.0.1", port=port)
+
+    @staticmethod
+    def make_qt_window(app_name: str, port: int):
+        """
+        Make Qt window and load dash url with the given port.
+
+        :param app_name: App name to display as Qt window title.
+        :param port: Port where the dash app has been launched.
+        """
+        app = QtWidgets.QApplication(sys.argv)
+        browser = QtWebEngineWidgets.QWebEngineView()
+
+        browser.setWindowTitle(app_name)
+        browser.load(QtCore.QUrl("http://127.0.0.1:" + str(port)))
+        # Brings Qt window to the front
+        browser.setWindowFlags(QtCore.Qt.WindowStaysOnTopHint)
+        # Setting window size
+        browser.resize(1200, 800)
+        browser.show()
+
+        app.exec_()  # running the Qt app
+        os.kill(os.getpid(), signal.SIGTERM)  # shut down dash server and notebook
+
+    def launch_qt(self, objects: str, n_clicks: int) -> str:  # pylint: disable=W0613
+        """
+        Launch the Qt app when launch app button is clicked.
+
+        :param objects: Selected object uid.
+        :param n_clicks: Number of times button has been clicked; triggers callback.
+
+        :return launch_app_markdown: Empty string since callbacks must have output.
+        """
+
+        triggers = [c["prop_id"].split(".")[0] for c in callback_context.triggered]
+        if "launch_app" in triggers and objects is not None:
+            # Make new workspace with only the selected object
+            obj = self.workspace.get_entity(uuid.UUID(objects))[0]
+
+            temp_geoh5 = "Scatterplot_" + f"{time():.0f}.geoh5"
```

Comment:
Let's generalize that and imply append a time stamp to the name of the incoming `workspace.h5file`